packages/epo-widget-lib/src/widgets/ColorTool/ImageComposite/ImageComposite.tsx

```diff
@@ -52,16 +70,16 @@ const ImageComposite = forwardRef<
           style={{ "--loading-opacity": isLoading ? 0 : 1 }}
         >
           {filters &&
-            filters.map((filter) => {
-              const { label, image, color, brightness, active } = filter;
+            filters.map((filter, i) => {
+              const { label, image, color, brightness } = filter;
 
               return (
-                <FilterImage
-                  key={`filter-${label}`}
+                <OffscreenFilter
```

Comment:
this is now a headless component (because hooks can't run in loops) that simple returns it's canvas on changes